tests/integration_tests/data_steward/cdr_cleaner/cleaning_rules/replace_standard_id_in_domain_tables_test.py

```diff
@@ -0,0 +1,252 @@
+            project_id=self.project_id, dataset_id=self.dataset_id)
+
+        concept_relationship_data_query = CONCEPT_RELATIONSHIP_DATA_TEMPLATE.render(
+            project_id=self.project_id, dataset_id=self.dataset_id)
+
+        condition_occurrence_data_query = CONDITION_OCCURRENCE_DATA_TEMPLATE.render(
+            project_id=self.project_id, dataset_id=self.dataset_id)
+
+        mapping_condition_occurrence_data_query = MAPPING_CONDITION_OCCURRENCE_DATA_TEMPLATE.render(
+            project_id=self.project_id, dataset_id=self.dataset_id)
+
+        # Load test data
+        self.load_test_data([
+            f'''{concept_data_query};
+                {concept_relationship_data_query};
+                {condition_occurrence_data_query};
+                {mapping_condition_occurrence_data_query}'''
+        ])
+
+    def test_replace_standard_id_in_domain_tables(self):
+        """
+        condition_occurrence 
+        
+        record 1: condition_concept_id is already standard, so do nothing 
+        
+        record 2: condition_concept_id is 0, but condition_source_concept_id 45567179 can be 
+        mapped to the standard concept 319835. So condition_concept_id should be replaced with 
+        319835 and this record is sandboxed.
+        
+        record 3: condition_concept_id is the non-standard 45567179 and 
+        condition_source_concept_id 0. 45567179 can be mapped to the standard concept 319835. So 
+        condition_concept_id should be replaced with 319835 and condition_source_concept_id 
+        should be replaced by the original condition_concept_id 45567179 and this record is 
+        sandboxed. 
+        
+        record 4: condition_concept_id is the non-standard 45567179 and 
+        condition_source_concept_id the same non-standard 45567179. 45567179 can be mapped to the 
+        standard concept 319835. So condition_concept_id should be replaced with 319835 and 
+        condition_source_concept_id remains the same and this record is sandboxed. 
+        
+        record 5: condition_concept_id is the non-standard 40398862 and 
+        condition_source_concept_id the same non-standard 40398862, but there is no standard 
+        concept mapping for 40398862. Although condition_concept_id and 
+        condition_source_concept_id remain the same, this record is still sandboxe because this 
+        record is flagged as being non-standard. 
+        
+        record 6: condition_concept_id is the non-standard 45587397 and 
+        condition_source_concept_id the same non-standard 45587397. 45567179 can be mapped to two 
+        standard concepts 434701 and 444094. So this record is split into two new records, 
+        the new ids are generated using this logic MAX(condition_occurrence_id) + ROW_NUMBER(). 
+        Two new records with condition_occurrence_id 7 and 8 are generated and the corresponding 
+        mapping table _mapping_condition_occurrence is updated as well. This record is sandboxed. 
```

Comment:
@nishanthpp93 this is a good point, I totally forgot about it. So like you said, the mapping table could track the provenance of the records and it should be okay to use the logic `max(condition_occurrence_id)`, however, it will make the `_mapping_domain_id` inconsistent in the mapping table because the HPO gets a designated constant or a "PARTITION" for generating the new IDs. So are there any analyses or functions depending on the fact that all the HPO records need to be assigned with the same constant?

Reply:
I'm personally not aware of any such instances, but maybe this needs to be brought up to the team.

Reply:
Yes sounds good! I remember (I could remember wrong) the reason we add a constant to each HPO site is due to the limitation of BigQuery, initially we tried to have BigQuery automatically generate new IDs for all records using ROW_NUMBER() but failed to do so, that's why we had to generate new IDs ourselves using a constant (constant + condition_occurrence_id).

Reply:
We are gonna discuss this at the Eng-Sync meeting